packages/eslint-plugin/src/rules/directive-class-suffix.ts

```diff
@@ -39,41 +39,40 @@ export default createESLintRule<Options, MessageIds>({
       },
     ],
     messages: {
-      directiveClassSuffix: `The name of the class {{className}} should end with the suffix {{suffixes}} (${STYLE_GUIDE_LINK})`,
+      directiveClassSuffix: `The name of the class {{className}} should end with suffix(es) {{suffixes}} (${STYLE_GUIDE_LINK})`,
     },
   },
-  defaultOptions: [
-    {
-      suffixes: ['Directive'],
-    },
-  ],
-  create(context, [options]) {
-    const { suffixes } = options;
-
+  defaultOptions: [{ suffixes: DEFAULT_SUFFIXES }],
+  create(context, [{ suffixes }]) {
     return {
       [DIRECTIVE_CLASS_DECORATOR](node: TSESTree.Decorator) {
+        const selectorPropertyValue = getDecoratorPropertyValue(
+          node,
+          'selector',
+        );
+
+        if (!selectorPropertyValue) return;
+
         const classParent = node.parent as TSESTree.ClassDeclaration;
         const className = getClassName(classParent);
-
         const declaredInterfaceNames = getDeclaredInterfaceNames(classParent);
         const hasValidatorInterface = declaredInterfaceNames.some(
-          (interfaceName) => interfaceName.endsWith(ValidatorSuffix),
+          (interfaceName) => interfaceName.endsWith(VALIDATOR_SUFFIX),
+        );
+        const allSuffixes = suffixes.concat(
+          hasValidatorInterface ? VALIDATOR_SUFFIX : [],
         );
-
-        if (hasValidatorInterface) {
-          suffixes.push(ValidatorSuffix);
-        }
 
         if (
           !className ||
-          !suffixes.some((suffix) => className.endsWith(suffix))
+          !allSuffixes.some((suffix) => className.endsWith(suffix))
         ) {
           context.report({
-            node: classParent.id ? classParent.id : classParent,
+            node: classParent.id ?? classParent,
             messageId: 'directiveClassSuffix',
             data: {
               className,
-              suffixes,
+              suffixes: getReadablePrefixes(allSuffixes),
```

Comment:
This is confusing now, please could we rename the `getReadablePrefixes` utility?

It is just a generic helper for listing some possible alternative values, so maybe we could call it:

`listPossibleValues()`

Reply:
The rename makes perfect sense! What do you say about `toHumanReadableText`?